src/lib/mocha/mocha-helper.js

```diff
@@ -16,3 +18,13 @@ after(function () {
     log.debug('[chimp][mocha-helper] Ended browser sessions');
   }
 });
+
+afterEach(function () {
+  if (screenshotHelper.shouldTakeScreenshot(this.currentTest.state)) {
+    if (booleanHelper.isTruthy(process.env['chimp.saveScreenshotsToDisk'])) {
+      const affix = this.currentTest.state !== 'passed' ? ' (failed)' : '';
+      const fileName = this.currentTest.title + affix;
```

Comment:
Because there can be many levels describe and it blocks, I think it might be better to use **this.currentTest.fullTitle()**. That said, both of these could suffer for a description of a test that is very detailed and there generally are issues with the filename being so long. I think we might want to chop off and limit the number of characters too.